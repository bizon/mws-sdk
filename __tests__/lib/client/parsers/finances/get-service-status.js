const parseXml = require('../../../../../lib/client/parsers')
const parseGetServiceStatusResponse = require('../../../../../lib/client/parsers/base/get-service-status-response')

describe('lib.client.parsers.base.get-service-status-response', () => {
  it('should parse the GetServiceStatusResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetServiceStatusResponse xmlns="http://mws.amazonservices.com/Finances/2015-05-01">
        <GetServiceStatusResult>
          <Status>GREEN</Status>
          <Timestamp>2015-05-01T18:12:21.687Z</Timestamp>
        </GetServiceStatusResult>
        <ResponseMetadata>
          <RequestId>082c41fd-2f6b-4616-a518-7db14EXAMPLE</RequestId>
        </ResponseMetadata>
      </GetServiceStatusResponse>`,
    )

    const res = parseGetServiceStatusResponse('/finances:GetServiceStatusResponse', doc)

    expect(res).toMatchSnapshot()
  })
})
