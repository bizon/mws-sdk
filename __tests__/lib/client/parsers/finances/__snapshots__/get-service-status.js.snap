// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.base.get-service-status-response should parse the GetServiceStatusResponse example response from MWS doc 1`] = `
Object {
  "getServiceStatusResult": Object {
    "status": "GREEN",
    "timestamp": 2015-05-01T18:12:21.687Z,
  },
  "responseMetadata": Object {
    "requestId": "082c41fd-2f6b-4616-a518-7db14EXAMPLE",
  },
}
`;
